messaging/kafka/src/main/java/io/helidon/messaging/kafka/connector/KafkaMessage.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.messaging.kafka.connector;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+/**
+ * Kafka specific MP messaging message.
+ *
+ * @param <K> kafka record key type
+ * @param <V> kafka record value type
+ */
+public class KafkaMessage<K, V> implements Message<ConsumerRecord<K, V>> {
+
+    private ConsumerRecord<K, V> consumerRecord;
+    private CompletableFuture<Void> ackFuture = new CompletableFuture<>();
+
+    /**
+     * Kafka specific MP messaging message.
+     *
+     * @param consumerRecord {@link org.apache.kafka.clients.consumer.ConsumerRecord}
+     */
+    public KafkaMessage(ConsumerRecord<K, V> consumerRecord) {
+        this.consumerRecord = consumerRecord;
+    }
+
+    @Override
+    public ConsumerRecord<K, V> getPayload() {
+        return consumerRecord;
+    }
+
+    public CompletableFuture<Void> getAckFuture() {
+        return ackFuture;
+    }
+
+    @Override
+    public CompletionStage<Void> ack() {
+        ackFuture.complete(null);
```

Comment:
I spoke with @danielkec about this. The thing is that polling from Kafka is a blocking operation, and in reactive streams we cannot have threads blocked. So we need some way to make it in non-blocking way.

The workaround to deal with this is the BackPressureLayer. There we are buffering events coming from polling. That KafkaMessage.ack() is communication with BackPressureLayer, instead of doing it with Kafka.